addons/project/models/project_project.py

```diff
@@ -428,7 +428,12 @@ def copy(self, default=None):
         default = dict(default or {})
         # Since we dont want to copy the milestones if the original project has the feature disabled, we set the milestones to False by default.
         default['milestone_ids'] = False
-        new_projects = super(Project, self.with_context(mail_auto_subscribe_no_notify=True, mail_create_nosubscribe=True)).copy(default=default)
+        self_with_context = self.with_context(
+            mail_auto_subscribe_no_notify=True,
+            mail_create_nosubscribe=True,
+            default_stage_id=self._default_stage_id().id
+        )
+        new_projects = super(Project, self_with_context).copy(default=default)
```

Comment:
I think it would be better to pop the context key, because the order of default is:

- `default_{field_name}` in context
- `ir.default` (called "user defined default" in the interface that a user can set to override default)
- default attribute on the field

so doing this change will prevent `ir.default` to work if there was one.

```suggestion
        copy_context = dict(
             self.env.context,
             mail_auto_subscribe_no_notify=True,
             mail_create_nosubscribe=True,
         )
         copy_context.pop("default_stage_id", None)
         new_projects = super(Project, self.with_context(copy_context)).copy(default=default)